### PR TITLE
Add support for customized missing field resolver handling

### LINKED ISSUE #129: Support for a catch-all field resolver
In my schema I support object synthetic properties via a **compute(expression: String!)** method which does its job allowing me to return to a query any number of results. GraphQL specification allows renaming them in the result set by specifying aliases in the query:
```javascript
positions
{
    house_shorts
    house_longs
    nominal_short: compute(expression: "self.house_shorts * self.price")
    nominal_long: compute(expression: "self.house_longs * self.price")
}
```
This is all good but the problem is that I would like to be able to reference these properties further away in the query:
```
stream
{
    positions
    {
        ...
        nominal_short(expression: "self.house_shorts * self.price)
        nominal_long(expression: "self.house_longs * self.price")
    }
    filter(expression: "self.positions.nominal_short > 1000000 or self.positions.nominal_long > 1000000")
    {
        list
        {
            nominal_short
            nominal_long
        }
    }
}
```
In the latter query **positions, nominal_** are properties created at runtime and they cannot be resolved by the current implementation of this library. However, if the library would have a MissingFieldResolver which instead of throwing an exception would simply allow for calling a user defined handler things could work out differently. The handler can use a Map to associate "missing" properties with their values. A type resolver could implement a method, tag it with an annotation @GenericFieldResolver and return the value from it.

### PR DESCRIPTION
**Pull request by timward60**
Resolves #129

## Checklist
- [X] Pull requests follows the [contribution guide](https://github.com/graphql-java-kickstart/graphql-java-tools/wiki/Contribution-guide)
- [X] New or modified functionality is covered by tests

## Description
This pull request adds support for custom missing field handling which allows the ability to have a catch-all field resolver implementation.

This feature is needed for our project to allow similar functionality to [GraphQL-Faker](https://github.com/APIs-guru/graphql-faker) where we want to return fake values for our clients teams to be able to experiment with our service schema.

To add custom missing resolver data fetcher you need to implement the following `DataFetcher<Any?>` interface:
```kotlin
class ExampleMissingResolverDataFetcher : DataFetcher<Any?> {
  override fun get(env: DataFetchingEnvironment?): Any? {
    return env?.getArgument("input")
  }
}
```

When building `SchemaParserOptions` you need to inject the custom missing resolver data fetcher:
```kotlin
SchemaParserOptions.newOptions()
  .missingResolverDataFetcher(ExampleMissingResolverDataFetcher())
  .build()
```

## Pending Work / Feedback Requested:
The following pending work remains and I am also requesting feedback here:
* Is this the right approach to allow catch-all field resolver?
* Some feedback on test coverage and if more unit tests should be added.
* Updating developer documentation.

### CODE REVIEW

File: src/main/kotlin/graphql/kickstart/tools/MissingFieldResolverHandler.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package graphql.kickstart.tools
+
+import graphql.schema.DataFetchingEnvironment
+
+interface MissingFieldResolverHandler {
```
Comment (vojtapol) [inline src/main/kotlin/graphql/kickstart/tools/MissingFieldResolverHandler.kt:5]: Any chance we could let the user pass in the whole `graphql.schema.DataFetcher`? I feel like this interface is the same thing and just adds one layer of indirection. Let me know your thoughts.

**Review by vojtapol — CHANGES_REQUESTED** [review]
Thank you for the contribution! This is an interesting addition that seems pretty easy to merge.

File: src/main/kotlin/graphql/kickstart/tools/MissingFieldResolverHandler.kt (hunk 1 above)
Comment (timward60) [reply src/main/kotlin/graphql/kickstart/tools/MissingFieldResolverHandler.kt]: Yep that makes sense, pushed another commit to allow the consumer to directly provide the `DataFetcher<Any?>`.

**Comment by timward60** [comment]
One thing I am wondering, if you can provide some guidance. Are there any suggested approaches to allow GraphQL Java Tools to redefine the description/or inject a SDL schema directive a field if it has a missing field resolver?

It would be great when the client executes the introspection query that we can prefix the description with PREVIEW: or add a SDL schema directive that we can extend our hosted GraphQL playground instance to highlight fields missing field resolvers. We would prefer not to have this automated rather than explicitly documented in our SDL files.

**Comment by timward60** [comment]
@vojtapol is there anything else pending for this to be merged?

**Comment by oryan-block** [comment]
Hi @timward60,
If we wanted to mark missing fields with a description we'd need to detect that their resolvers are `MissingFieldResolver` during object field creation in the `SchemaParser` and update the description there. 
To do anything more involved or customizable would probably require another schema option field of some kind. 
However, I would argue that that might be an overkill at this point.

**Review by vojtapol — APPROVED** [review]

**Pull request merged**